This patch changes the SSC platform configuration on all ST 710x boards such
that the SSC port containing the HDMI detect line is I2C only. 

This is because on all of the affected boards the HDMI detect is wired into
the same pin as would be used by an SPI data line.  As the SSC port is only
brought out to a test header, it is not available for normal use and hence
only the two-wire I2C bus is available.

The patch also rewrites the SSC configuration setup to use the symbolic 
constants, rather than just a hex number, so the configuration is human 
readable.

Signed off by: Stephen Gallimore <stephen.gallimore@st.com>
Index: linux/arch/sh/boards/st/stb7100mboard/setup.c
===================================================================
--- linux.orig/arch/sh/boards/st/stb7100mboard/setup.c	2007-08-30 11:04:41.932896000 +0100
+++ linux/arch/sh/boards/st/stb7100mboard/setup.c	2007-08-30 11:20:06.682592000 +0100
@@ -186,13 +186,16 @@
 };
 
 static struct plat_ssc_pio_t ssc_pio[] = {
-        {2, 0, 2, 1, 2, 2},
-        {3, 0, 3, 1, 3, 2},
-        {4, 0, 4, 1, 0xff, 0xff},
+	{2, 0, 2, 1, 0xff, 0xff},
+	{3, 0, 3, 1, 3, 2},
+	{4, 0, 4, 1, 0xff, 0xff},
 };
 static struct plat_ssc_data ssc_private_info = {
-         .capability  = 0x1f,
-         .pio         = ssc_pio
+	.capability  =
+		(SSC_I2C_CAPABILITY << (0*2)) |
+		((SSC_SPI_CAPABILITY | SSC_I2C_CAPABILITY) << (1*2)) |
+		(SSC_I2C_CAPABILITY << (2*2)),
+	.pio         = ssc_pio
 };
 struct platform_device ssc_device = {
         .name = "ssc",
Index: linux/arch/sh/boards/st/stb7100ref/setup.c
===================================================================
--- linux.orig/arch/sh/boards/st/stb7100ref/setup.c	2007-08-30 11:04:41.940896000 +0100
+++ linux/arch/sh/boards/st/stb7100ref/setup.c	2007-08-30 11:20:06.690592000 +0100
@@ -181,12 +181,15 @@
 };
 
 static struct plat_ssc_pio_t ssc_pio[] = {
-	{2, 0, 2, 1, 2, 2},
+	{2, 0, 2, 1, 0xff, 0xff},
 	{3, 0, 3, 1, 3, 2},
 	{4, 0, 4, 1, 0xff, 0xff},
 };
 static struct plat_ssc_data ssc_private_info = {
-	.capability  = 0x1b,
+	.capability  =
+		(SSC_I2C_CAPABILITY << (0*2)) |
+		(SSC_SPI_CAPABILITY << (1*2)) |
+		(SSC_I2C_CAPABILITY << (2*2)),
 	.pio         = ssc_pio
 };
 struct platform_device ssc_device = {
Index: linux/arch/sh/boards/st/stb7109eref/setup.c
===================================================================
--- linux.orig/arch/sh/boards/st/stb7109eref/setup.c	2007-08-30 11:04:41.948896000 +0100
+++ linux/arch/sh/boards/st/stb7109eref/setup.c	2007-08-30 11:20:06.697592000 +0100
@@ -175,13 +175,16 @@
 };
 
 static struct plat_ssc_pio_t ssc_pio[] = {
-	{2, 0, 2, 1, 2, 2},
+	{2, 0, 2, 1, 0xff, 0xff},
 	{3, 0, 3, 1, 3, 2},
 	{4, 0, 4, 1, 0xff, 0xff},
 };
 static struct plat_ssc_data ssc_private_info = {
-	 .capability  = 0x1f,
-	 .pio	      = ssc_pio
+	.capability  =
+		(SSC_I2C_CAPABILITY << (0*2)) |
+		((SSC_SPI_CAPABILITY | SSC_I2C_CAPABILITY) << (1*2)) |
+		(SSC_I2C_CAPABILITY << (2*2)),
+	.pio         = ssc_pio
 };
 struct platform_device ssc_device = {
 	.name = "ssc",
@@ -192,7 +195,6 @@
 		 .platform_data = &ssc_private_info
 	}
 };
-
 static struct plat_lirc_data lirc_private_info = {
 	/* For the 7100, the clock settings will be calculated by the driver
 	 * from the system clock
Index: linux/arch/sh/boards/st/hms1/setup.c
===================================================================
--- linux.orig/arch/sh/boards/st/hms1/setup.c	2007-08-30 11:04:41.923896000 +0100
+++ linux/arch/sh/boards/st/hms1/setup.c	2007-08-30 11:20:06.705592000 +0100
@@ -138,12 +138,15 @@
 };
 
 static struct plat_ssc_pio_t ssc_pio[] = {
-	{2, 0, 2, 1, 2, 2},
+	{2, 0, 2, 1, 0xff, 0xff},
 	{3, 0, 3, 1, 3, 2},
 	{4, 0, 4, 1, 0xff, 0xff},
 };
 static struct plat_ssc_data ssc_private_info = {
-	.capability  = 0x1f,
+	.capability  =
+		(SSC_I2C_CAPABILITY << (0*2)) |
+		((SSC_SPI_CAPABILITY | SSC_I2C_CAPABILITY) << (1*2)) |
+		(SSC_I2C_CAPABILITY << (2*2)),
 	.pio         = ssc_pio
 };
 struct platform_device ssc_device = {
